pkg/controller/hibernation/hibernation_controller_test.go

```diff
@@ -392,10 +392,11 @@ func TestReconcile(t *testing.T) {
 		},
```

Comment:
The above test is hibernating based on `hibernateAfter`, so it's not hitting the new chunk of code referenced previously. We should add a new test with a CD with:
- *no* `hibernateAfter`
- `powerState=Hibernating`
- the hibernating condition `False` or `Unknown` (with whatever `Reason`)
and assert that the final `powerState` and condition represent hibernating.

Reply:
Added - actually modified an existing one since there are other tests for hibernateafter. Also, when hibernateafter is not set, we change the powerstate, so an assert on that would be not necessary

Reply:
> when hibernateafter is not set, we change the powerstate, so an assert on that would be not necessary

I would think that would be a really good reason to include the assertion :)